onlineShop.core/Managers/CartsManager.cs

```diff
@@ -0,0 +1,178 @@
+
+            Dictionary<Guid, Guid> dictionaryOfReservedProducts = new Dictionary<Guid, Guid>(); // productId and reservationId
+            foreach (var reservation in listOfReservations)
+            {
+                dictionaryOfReservedProducts.Add(reservation.ProductId, reservation.Id);
+                foreach (var item in cart.Products)
+                {
+                    if (dictionaryOfReservedProducts.ContainsKey(item))
+                    {
+                        _reservationsManager.TryRenewReservation(dictionaryOfReservedProducts[item]);
+                        dictionaryOfReservedProducts.Remove(item);
+                    }
+                    else
+                    {
+                        Reservation newReservation;
+                        var product = _productsProvider.GetProducts().First(id => id.Id == item);
+                        _reservationsManager.TryToReserveProduct(product, cart.SessionId, out newReservation);
+                    }
+                    break;
+                }
+            }
+            listOfReservations = (List<Reservation>)_reservationsProvider.GetReservations().Where(sId => sId.SessionId == cart.SessionId);
+            cart.Products = new List<Guid>();
+            foreach (var reservation in listOfReservations)
+            {
+                cart.Products.Add(reservation.ProductId);
+            }
+
+            _cartsProvider.SaveCart(cart);
+        }
+
+        public Dictionary<Product, int> GetProductsCount(Cart cart)
+        {
+            Dictionary<Product, int> cartToDisplay = new Dictionary<Product, int>();
+
+            foreach (var item in cart.Products)
+            {
+                var product = _productsProvider.GetProducts().First(id => id.Id == item);
+                if (!cartToDisplay.ContainsKey(product))
+                {
+                    cartToDisplay.Add(product, 1);
+                }
+                else
+                {
+                    cartToDisplay[product]++;
+                }
+            }
+
+            return cartToDisplay;
+        }
+
+        public double GetTotalPrice(Dictionary<Product, int> products)
+        {
+            double totalPrice = 0;
+            foreach (var item in products)
+            {
+                totalPrice += item.Key.Price * item.Value;
+            }
+
+            return totalPrice;
+        }
+
+        public double GetTotalWeight(Dictionary<Product, int> products)
+        {
+            double totalWeight = 0;
+            foreach (var item in products)
+            {
+                if (item.Key is PhysicalProduct)
+                {
+                    PhysicalProduct physicalItem = item.Key as PhysicalProduct;
+                    totalWeight += physicalItem.Size.Weight * item.Value;
+                    return totalWeight;
+                }
+            }
+
+            return 0;
+        }
+
+        //public void CompletePurchase(Cart cart)
```

Comment:
Why is this code commented? Also there are empty lines at the end

Reply:
Fixed